main.cpp

```diff
@@ -14,27 +14,35 @@ using namespace saneql;
 //---------------------------------------------------------------------------
 // (c) 2023 Thomas Neumann
 //---------------------------------------------------------------------------
-static string readFile(const string& fileName) {
+static void readFile(const string& fileName, ostringstream& output) {
    ifstream in(fileName);
    if (!in.is_open()) {
       cerr << "unable to read " << fileName << endl;
       exit(1);
    }
-   ostringstream str;
-   str << in.rdbuf();
-   return str.str();
+   output << in.rdbuf();
+}
+//---------------------------------------------------------------------------
+static string readSaneQL(const string& saneqlFile, std::optional<string> dialectFile = {}) {
+   ostringstream output;
+   if (dialectFile) {
+      output << "\n";
+      readFile(*dialectFile, output);
+   }
+   readFile(saneqlFile, output);
+   return output.str();
 }
 //---------------------------------------------------------------------------
 int main(int argc, char* argv[]) {
-   if (argc != 2) {
-      cerr << "usage: " << argv[0] << " file" << endl;
+   if (argc < 2) {
+      cerr << "usage: " << argv[0] << " file [dialect-file]" << endl;
```

Comment:
I think that is surprising behavior. Instead of having an optional second argument that is then read first, the program should simply read all input files in sequence and append them implicitly. That covers your use case and is much more intuitive.

Reply:
Yes, that is much better. I changed the behaviour.